cap/dyn_supervisor.go

```diff
@@ -9,36 +9,141 @@ import (
 	"github.com/capatazlib/go-capataz/internal/c"
 )
 
+// ctrlMsg executes some control logic on the Supervisor goroutine
```

Comment:
I am glad you did this. I think it is easier to follow. I thought you might do this in a different PR.